Add websocket denial test

desimone opened the pull request:
## Summary
- fix err variable check in websocket message read
- add `TestHttp1WebsocketDenied` to ensure websockets are rejected when not allowed

## Related issues

## User Explanation
Adds a regression test verifying that routes without `AllowWebsockets` return an error when establishing websocket connections.

## Checklist
- [ ] reference any related issues
- [x] updated unit tests
- [ ] add appropriate label (`enhancement`, `bug`, `breaking`, `dependencies`, `ci`)
- [ ] ready for review

desimone committed:
config: add websocket denied test

desimone committed:
Merge branch 'main' into bdd/fix-websocket-read-error-handling

kenjenkins commented on config/envoyconfig/protocols_int_test.go:
```diff
@@ -352,6 +352,36 @@ func TestHttp1Websocket(t *testing.T) {
+	snippets.WaitStartupComplete(env)
+
+	err := up.DialWS(route, func(_ *websocket.Conn) error { return nil }, upstreams.Path("/ws"))
+	require.Error(t, err)
```
Tests that exercise failure modes are often tricky, because it's easy for the test to fail for the wrong reason.

Is there a specific error message we should look for here?

Or, even better, should we merge this scenario into the `TestHttp1Websocket` method above? We could define a second route to the same upstream, just without the `AllowWebsockets` option set. That way we would know that the `DialWS()` error isn't because of any misconfiguration of the upstream server.

desimone committed:
test(envoyconfig): fold websocket denial into TestHttp1Websocket

Per review feedback, replace the standalone TestHttp1WebsocketDenied
(which asserted only that DialWS returned some error, and so could pass
for an unrelated reason) with allowed/denied subtests that share one
upstream and websocket handler and differ only in the AllowWebsockets
policy flag. The allowed subtest proves the upstream works, so a failure
on the denied subtest is attributable to policy. Assert the specific
websocket.ErrBadHandshake on denial instead of a bare error.

Addresses kenjenkins review comment on #5622.
